core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -1126,6 +1132,200 @@ public OptimizelyConfig getOptimizelyConfig() {
+
+        DecisionNotification decisionNotification = DecisionNotification.newFlagDecisionNotificationBuilder()
+            .withUserId(userId)
+            .withAttributes(copiedAttributes)
+            .withFlagKey(key)
+            .withEnabled(flagEnabled)
+            .withVariables(variableMap)
+            .withVariationKey(variationKey)
+            .withRuleKey(ruleKey)
+            .withReasons(reasonsToReport)
+            .withDecisionEventDispatched(decisionEventDispatched)
+            .build();
+        notificationCenter.send(decisionNotification);
+
+        logger.info("Feature \"{}\" is enabled for user \"{}\"? {}", key, userId, flagEnabled);
+
+        return new OptimizelyDecision(
+            variationKey,
+            flagEnabled,
+            optimizelyJSON,
+            ruleKey,
+            key,
+            user,
+            reasonsToReport);
+    }
+
+    Map<String, OptimizelyDecision> decideForKeys(@Nonnull OptimizelyUserContext user,
+                                                  @Nonnull List<String> keys,
+                                                  @Nonnull List<OptimizelyDecideOption> options) {
+        Map<String, OptimizelyDecision> decisionMap = new HashMap<>();
+
+        ProjectConfig projectConfig = getProjectConfig();
+        if (projectConfig == null) {
+            logger.error("Optimizely instance is not valid, failing isFeatureEnabled call.");
+            return decisionMap;
+        }
+
+        if (keys.isEmpty()) return decisionMap;
+
+        List<OptimizelyDecideOption> allOptions = getAllOptions(options);
+
+        for (String key : keys) {
+            OptimizelyDecision decision = decide(user, key, options);
+            if (!allOptions.contains(OptimizelyDecideOption.ENABLED_FLAGS_ONLY) || decision.getEnabled()) {
+                decisionMap.put(key, decision);
+            }
+        }
+
+        return decisionMap;
+    }
+
+    Map<String, OptimizelyDecision> decideAll(@Nonnull OptimizelyUserContext user,
+                                              @Nonnull List<OptimizelyDecideOption> options) {
+        Map<String, OptimizelyDecision> decisionMap = new HashMap<>();
+
+        ProjectConfig projectConfig = getProjectConfig();
+        if (projectConfig == null) {
+            logger.error("Optimizely instance is not valid, failing isFeatureEnabled call.");
+            return decisionMap;
+        }
+
+        List<FeatureFlag> allFlags = projectConfig.getFeatureFlags();
+        List<String> allFlagKeys = new ArrayList<>();
+        for (int i = 0; i < allFlags.size(); i++) allFlagKeys.add(allFlags.get(i).getKey());
+
+        return decideForKeys(user, allFlagKeys, options);
+    }
+
+    private List<OptimizelyDecideOption> getAllOptions(List<OptimizelyDecideOption> options) {
+        List<OptimizelyDecideOption> copiedOptions = new ArrayList(defaultDecideOptions);
+        copiedOptions.addAll(options);
```

Comment:
As it would throw null pointer exception if options is null. 
```suggestion
        if (options != null) {
            copiedOptions.addAll(options);
        }
```